src/main/java/io/github/triglav_dataflow/jenkins/trigger/polling_triglav/AlternativeExecutionTime.java

```diff
@@ -0,0 +1,49 @@
+package io.github.triglav_dataflow.jenkins.trigger.polling_triglav;
+
+import java.util.Calendar;
+import java.util.Date;
+
+public class AlternativeExecutionTime
+{
+    private AlternativeExecutionTime()
+    {
+    }
+
+    public static final boolean shouldBuild(Date alternativeExecutionTime, JenkinsJob jenkinsJob, TimeUnit timeUnit)
+    {
+        // timeUnit is hourly, daily
+        // alternativeExecutionTime is set
+        // alternativeExecutionTime has passed
+        if (!TimeUnit.SINGULAR.equals(timeUnit) &&
+            null != alternativeExecutionTime &&
+            alternativeExecutionTime.before(new Date())) {
```

Comment:
This conditions are a bit complicated, so it's better to divide these like below.

```
if (TimeUnit.SINGULAR.equals(timeUnit) || alternativeExecutionTime == null) {
    // insufficient requirements
    return false;
}

if (alternativeExecutionTime.after(new Date())) {
    // alternativeExecutionTime has not passed
    return false;
}

...
```

Reply:
Sure, I agree with you